src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariationDiscoveryPipelineSpark.java

```diff
@@ -0,0 +1,177 @@
+            = new StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection();
+
+
+    @Override
+    public boolean requiresReads()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean requiresReference() {return true;}
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx ) {
+
+        final SAMFileHeader header = getHeaderForReads();
+        final PipelineOptions pipelineOptions = getAuthenticatedGCSOptions();
+
+        // gather evidence, run assembly, and align
+        final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList =
+                FindBreakpointEvidenceSpark
+                        .gatherEvidenceAndWriteContigSamFile(ctx, pipelineOptions, reAlignmentStageArgs, header, getUnfilteredReads(),
+                                outputSAM, localLogger);
+        if (alignedAssemblyOrExcuseList.isEmpty()) return;
+
+        // parse the contig alignments and extract necessary information
+        @SuppressWarnings("unchecked")
+        final JavaRDD<AlignedAssembly.AlignedContig> parsedAlignments = new InMemoryAlignmentParser().loadAndFilterAndParseContigAlignments(ctx, localLogger, header, alignedAssemblyOrExcuseList);
+        if(parsedAlignments.isEmpty()) return;
+
+        // discover variants and write to vcf
+        DiscoverVariantsFromContigAlignmentsSAMSpark
+                .discoverVariantsAndWriteVCF(parsedAlignments, discoverStageArgs.fastaReference,
+                        ctx.broadcast(getReference()), pipelineOptions, vcfOutputFileName, localLogger);
+    }
+
+    public static final class InMemoryAlignmentParser implements AssemblyAlignmentParser {
+        private static final long serialVersionUID = 1L;
+
+        @SuppressWarnings("unchecked")
+        public JavaRDD<AlignedAssembly.AlignedContig> loadAndFilterAndParseContigAlignments(final JavaSparkContext ctx,
+                                                                                            final Logger toolLogger,
+                                                                                            Object... localArguments) {
+
+            final SAMFileHeader header = (SAMFileHeader) localArguments[0];
+            final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList = (List<AlignedAssemblyOrExcuse>) localArguments[1];
+
+            // here we have two options, one is going through the route "BwaMemAlignment -> SAM -> GATKRead -> SAM -> AlignmentInterval"
+            //                           which is the route if the discovery pipeline is run by "FindBreakpointEvidenceSpark -> write sam file -> load sam file -> DiscoverVariantsFromContigAlignmentsSAMSpark"
+            //                         , the other is to go directly "BwaMemAlignment -> AlignmentInterval", which is faster but not used here.
+            return ctx.parallelize( viaSAMRoute(alignedAssemblyOrExcuseList, header, toolLogger) );
```

Comment:
It strikes me that at one point we already had an RDD of AlignedAssemblyOrExcuse (I guess most recently in the code path in `FindBreakpointEvidenceSpark.handleAssemblies`) that we could just operate on if we could return it from that method (ie instead of returning a `List<AlignedAssemblyOrExcuse>` the FBE code would return an `RDD<AlignedAssemblyOrExcuse>`. Did you think about doing that at all? It seems like it would be more efficient than re-parallelizing the collection.